handbook/python-dataplugins/python-dataplugins.md

```diff
@@ -0,0 +1,310 @@
+# Python DataPlugins
+
+Create a DataPlugin to load, to register, or search your own custom file formats in LabVIEW or DIAdem, or to index, browse, and find your file formats with SystemLink DataFinder.
+
+You can also create DataPlugins using C++, VBS, LabVIEW or Python.
+
+!!! note "Note"
+    Browse through 230+ DataPlugins in the [NI Store](https://search.ni.com/nisearch/app/main/p/ap/global/lang/en/pg/1/sn/ssnav:dpl/) to check whether there is an existing DataPlugin for your data format.
+
+## Getting Started
+
+Python DataPlugins consist of only one Python file that contains all the logic. Almost all language features of the official Python 3.5.9 and its base libraries can be used.
+
+You can create Python DataPlugins in your favorite editor. Recommended:
+
+- [DIAdem](https://www.ni.com/en-us/shop/data-acquisition-and-control/application-software-for-data-acquisition-and-control-category/what-is-diadem.html) >= 2020
+- VSCode with the [NI Python DataPlugin Extension](https://github.com/ni/vscode-ni-python-dataplugins)
+
+### Plugin Class
+
+Start writing your DataPlugin by implementing
+
+```python
+class Plugin:
+```
+
+The class name cannot be changed.
+
+### Read Store
+
+You need to implement a `read_store` method in every Python DataPlugin. This method is called by DIAdem, LabVIEW, or SystemLink DataFinder when attempting to open your data file. The applications pass a set of useful parameters that can be accessed by the parameter array.
+
+<!-- markdownlint-disable -->
+<details>
+<summary>Example Code</summary>
+<!-- markdownlint-enable -->
+
+```python
+import datetime
+import os
+from pathlib import Path
+def read_store(self, parameter):
+   """
+      Reads data file and returns a Python dictionary

```

Comment:
do we need the plural here "data files"?

Reply:
it's one file so I would stick with singular